### PR TITLE
[PWA-2169] Query for available Dynamic Blocks

### PR DESCRIPTION
**Pull request by mikhaelbois**
## Description

Dynamic Block components for Venia and Page Builder (Adobe Commerce only).

## Related Issue

Closes https://jira.corp.magento.com/browse/PWA-2169.

#### Test scenario(s) for direct fix/feature

Note this will only work with an Adobe Commerce backend.
You also need the code from https://github.com/magento-commerce/magento2-pwa-commerce/pull/2 for the content to be rendered properly.

1. Setup two Customer Segment's, one for Registered Users and one for Guests
2. Create a Dynamic Block with no segment selected
3. Create a Dynamic Block for RU segment
4. Create a Dynamic Block for Guest segment
5. Add the Dynamic Blocks to a CMS Page
6. Test CMS Page as Guest
7. Test CMS Page as Customer

#### Any ad-hoc/edge case scenarios that need to be considered?

- Dynamic Blocks should update when signing in and signing out.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by pedrochiossi — APPROVED** [review]
LGTM well done @mikhaelbois 👍🏻

**Comment by mikhaelbois** [comment]
We need validation for this backend PR to move things along.
https://github.com/magento-commerce/magento2-pwa-commerce/pull/2

File: packages/peregrine/lib/talons/SignIn/useSignIn.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const useSignIn = props => {
 
                 // Clear all cart/customer data from cache and redux.
                 await apolloClient.clearCacheData(apolloClient, 'cart');
-                await apolloClient.clearCacheData(apolloClient, 'checkout');
+                await apolloClient.clearCacheData(apolloClient, 'customer');
```
Comment (jeremycharron) [inline packages/peregrine/lib/talons/SignIn/useSignIn.js:69]: I'm not sure to understand this change. Is cache data of checkout also in customer?

File: packages/venia-ui/lib/components/CmsDynamicBlock/__tests__/cmsDynamicBlock.ee.spec.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                variables: {
+                    type: DYNAMIC_BLOCK_FIXED_TYPE,
+                    locations: mockLocations,
+                    uids: mockUids
+                }
+            })
+        );
+    });
+
+    it('calls useQuery with locations', () => {
```
Comment (jeremycharron) [inline packages/venia-ui/lib/components/CmsDynamicBlock/__tests__/cmsDynamicBlock.ee.spec.js:218]: This test seems exactly the same as test on line 201. It is intended?

File: packages/venia-ui/lib/components/CmsDynamicBlock/cmsDynamicBlock.ee.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import ErrorView from '@magento/venia-ui/lib/components/ErrorView';
+
+import DynamicBlock from './dynamicBlock';
+
+export const DYNAMIC_BLOCK_FIXED_TYPE = `SPECIFIED`;
+export const DYNAMIC_BLOCK_SALES_RULE_TYPE = `CART_PRICE_RULE_RELATED`;
+export const DYNAMIC_BLOCK_CATALOG_RULE_TYPE = `CATALOG_PRICE_RULE_RELATED`;
+
+const getDynamicBlockType = displayMode => {
+    if (displayMode === 'fixed') {
```
Comment (jeremycharron) [inline packages/venia-ui/lib/components/CmsDynamicBlock/cmsDynamicBlock.ee.js:14]: Maybe we could have constant variable for `fixed`, `salesrule `and `catalogrule`?

**Review by jeremycharron — COMMENTED** [review]
Good job. I left 3 comments/questions
Also made a preliminary QA while working on PWA-2777, I can display dynamic blocks successfully.

File: packages/peregrine/lib/talons/SignIn/useSignIn.js (hunk 1 above)
Comment (mikhaelbois) [reply packages/peregrine/lib/talons/SignIn/useSignIn.js:69]: 'checkout' does not exist. I must have done the wrong copy paste back then.
It replaces this: https://github.com/magento/pwa-studio/blob/v12.0.0/packages/peregrine/lib/talons/SignIn/useSignIn.js#L71

**Review by jeremycharron — DISMISSED** [review]
Commit a104f1d resolves my comments.

**Comment by ericeoeur** [comment]
1. The Cypress testing for both `smokeTest.spec.js` and `map.spec.js` have failed. Results: snapshots did not match. 
2. GraphQL Calls - While on the browser displays the appropriate segmented dynamic blocks per segment (i.e. guest and Registered User), graphQL calls are only returning no segment and guest segment data. This has been discussed with @mikhaelbois for review.

**Comment by mikhaelbois** [comment]
@ericeoeur 
I've just tested the following and the tests works.
How did you manage to get failures? 
`yarn test:headless --base-url https://pr-3587.pwa-venia.com/ -s src/tests/integrationTests/pageBuilder/map.spec.js`
`yarn test:headless --base-url https://pr-3587.pwa-venia.com/ -s src/tests/integrationTests/pageBuilder/smokeTest.spec.js`

**Comment by ericeoeur** [comment]
> @ericeoeur I've just tested the following and the tests works. How did you manage to get failures? `yarn test:headless --base-url https://pr-3587.pwa-venia.com/ -s src/tests/integrationTests/pageBuilder/map.spec.js` `yarn test:headless --base-url https://pr-3587.pwa-venia.com/ -s src/tests/integrationTests/pageBuilder/smokeTest.spec.js`

Hi @mikhaelbois, as previously discussed I ran into these issues running on my cloud environment. However, I have ran these tests form the deployed PR and all tests are passing as stated.

**Review by ericeoeur — APPROVED** [review]
LGTM

**Comment by ericeoeur** [comment]
This has passed QA and will be merging soon. Thanks!
